addons/base_automation/models/base_automation.py

```diff
@@ -341,6 +341,8 @@ def write(self, vals, **kw):
                     old_vals.pop('id'): old_vals
                     for old_vals in (records.read(list(vals)) if vals else [])
                 }
+                # redefining records such that records with <NewId> are handled too
+                records = self.with_env(actions.env)
                 # call original method
                 write.origin(records, vals, **kw)
```

Comment:
```suggestion
                write.origin(self.with_env(actions.env), vals, **kw)
```
because records is already used for different recordsets above (with filtered out NewID) and below (with filtered out NewID and filtered out records by pre/post-conditions), so using it a third time complicates this a little I think.